### PR TITLE
STY: PEP8

### PR DESCRIPTION
**Pull request by CJ-Wright**
Please note this commit is not responsible for making certain code works, is good, etc. This is only aimed at PEP8.

This is also a massive work in progress.

### CODE REVIEW

File: doc/conf.py [hunk 1]
```diff
@@ -20,12 +20,12 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#sys.path.insert(0, os.path.abspath('.'))
+# sys.path.insert(0, os.path.abspath('.'))
```
Comment (chiahaoliu) [inline doc/conf.py]: Do you use a automated style check?

Content in this ``conf.py`` was generated by ``sphinx`` configuration and I would prefer to let it stay as it is due to this nature.

File: xpdacq/beamtime.py [hunk 2]
```diff
@@ -150,17 +152,18 @@ def tseries(dets, exposure, delay, num, *, md = None):
...
 def _nstep(start, stop, step_size):
-    ''' helper function to compute number of steps and step_size
-    '''
+    """ helper function to compute number of steps and step_size
+    """
     requested_nsteps = abs((start - stop) / step_size)
 
-    computed_nsteps = int(requested_nsteps)+1 # round down for finer step size
+    computed_nsteps = int(
+        requested_nsteps) + 1  # round down for finer step size
```
Comment (chiahaoliu) [inline xpdacq/beamtime.py]: I think this line is fine, isn't it? (less than 72 characters)

File: xpdacq/beamtime.py [hunk 3]
```diff
@@ -209,8 +212,9 @@ def register_scanplan(self, scanplan):
         if scanplan.short_summary() not in sp_name_list:
             self.scanplans.append(scanplan)
         else:
-            old_obj = [ obj for obj in self.scanplans
-                      if obj.short_summary() ==scanplan.short_summary()].pop()
+            old_obj = [obj for obj in self.scanplans
+                       if
+                       obj.short_summary() == scanplan.short_summary()].pop()
             old_obj_ind = self.scanplans.index(old_obj)
```
Comment (chiahaoliu) [inline xpdacq/beamtime.py]: Do we need to separate conditional statement every time? I haven't noticed a strict style in PEP 8

File: xpdacq/calib.py [hunk 4]
```diff
@@ -40,11 +41,14 @@ def _check_obj(required_obj_list):
                             "namespace".format(obj_str))
     return
 
+
 def _timestampstr(timestamp):
-    ''' convert timestamp to strftime formate '''
-    timestring = datetime.datetime.fromtimestamp(float(timestamp)).strftime('%Y%m%d-%H%M')
+    """ convert timestamp to strftime formate """
+    timestring = datetime.datetime.fromtimestamp(float(timestamp)).strftime(
+        '%Y%m%d-%H%M')
```
Comment (chiahaoliu) [inline xpdacq/calib.py]: align with opening delimiter?

File: xpdacq/glbl.py [hunk 5]
```diff
@@ -94,23 +96,24 @@ def unstage(self):
...
+    YAML_DIR,
+    CONFIG_BASE,
+    SAMPLE_DIR,
+    EXPERIMENT_DIR,
+    SCANPLAN_DIR,
+    TIFF_BASE,
+    USERSCRIPT_DIR,
+    IMPORT_DIR,
+    ANALYSIS_DIR
+]
```
Comment (chiahaoliu) [inline xpdacq/glbl.py]: align with opening delimiter?

File: xpdacq/tests/test_beamtimeSetup.py [hunk 6]
```diff
@@ -176,11 +173,12 @@ def test_import_userScript_Etc(self):
         # create archive file
         for el in tared_list:
             f_path = os.path.join(src, el)
-            open(f_path,'a').close()
+            open(f_path, 'a').close()
         cwd = os.getcwd()
-        os.chdir(src) # inevitable step for compression
+        os.chdir(src)  # inevitable step for compression
         tar_name = 'HappyMeal'
-        shutil.make_archive(tar_name,'tar') # now data should be in xpdUser/Import/
```
Comment (chiahaoliu) [inline xpdacq/tests/test_beamtimeSetup.py]: single line?

File: xpdacq/tests/test_beamtimeSetup.py [hunk 7]
```diff
@@ -231,5 +228,6 @@ def test_export_userScriptsEtc(self):
                                                    config_base_tail)))
         # are files in unpacked dirs?
         self.assertTrue('script.py' in os.listdir(os.path.join(glbl.home,
-                                                  userScript_dir_tail)))
-        self.assertTrue('touched.yml' in os.listdir(os.path.join(glbl.home, config_base_tail, yaml_dir_tail)))
+                                                               userScript_dir_tail)))
+        self.assertTrue('touched.yml' in os.listdir(
+            os.path.join(glbl.home, config_base_tail, yaml_dir_tail)))
```
Comment (chiahaoliu) [inline xpdacq/tests/test_beamtimeSetup.py]: align with opening delimiter?

File: xpdacq/utils.py [hunk 8]
```diff
@@ -263,16 +281,16 @@ def create_yaml(self, bt):
                         phase_dict = v
                     finally:
                         parsed_sa_md.update({'sample_composition':
-                                             composition_dict})
+                                                 composition_dict})
```
Comment (chiahaoliu) [inline xpdacq/utils.py]: align with opening delimiter?

File: xpdacq/utils.py [hunk 9]
```diff
@@ -263,16 +281,16 @@ def create_yaml(self, bt):
...
 
                         parsed_sa_md.update({'sample_phase':
-                                             phase_dict})
+                                                 phase_dict})
```
Comment (chiahaoliu) [inline xpdacq/utils.py]: align with opening delimiter?

File: xpdacq/utils.py [hunk 10]
```diff
@@ -392,24 +409,26 @@ def _phase_parser(self, phase_str):
...
-                com, amount = el.split(':') # expect [<com>, <amount>]
-            phase_dict.update({com.strip():float(amount.strip())})
+                com, amount = el.split(':')  # expect [<com>, <amount>]
+            phase_dict.update({com.strip(): float(amount.strip())})
             parsed_tuple = composition_analysis(com.strip())
             # expect: ([elment_1, element_2, ...], [sto_1, sto_2,...])
             for i in range(len(parsed_tuple[0])):
                 composition_dict.update({parsed_tuple[0][i]:
-                                         parsed_tuple[1][i]})
+                                             parsed_tuple[1][i]})
```
Comment (chiahaoliu) [inline xpdacq/utils.py]: align with opening delimiter?

File: xpdacq/xpdacq.py [hunk 11]
```diff
@@ -106,13 +106,15 @@ def periodic_dark(plan):
...
-        #print('qualified_dark_uid is {}'.format(qualified_dark_uid))
+        # print('qualified_dark_uid is {}'.format(qualified_dark_uid))
         # FIXME: should we do "or" or "and"?
-        if ((not need_dark) and (not qualified_dark_uid)):
+        if (not need_dark) and (not qualified_dark_uid):
             need_dark = True
-        if need_dark and (not qualified_dark_uid) and msg.command == 'open_run' and ('dark_frame' not in msg.kwargs):
+        if need_dark and (
+                not qualified_dark_uid) and msg.command == 'open_run' and (
+                    'dark_frame' not in msg.kwargs):
```
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: align with opening delimiter?

File: xpdacq/xpdacq.py [hunk 12]
```diff
@@ -150,14 +151,15 @@ def _validate_dark(expire_time=None):
...
-                         abs(el['exposure'] - light_cnt_time) <= acq_time and
-                         abs(el['timestamp'] - now) <= (expire_time*60 - acq_time)
-                         and (el['acq_time'] == acq_time)
-                         ]
+    qualified_dark_uid = [el['uid'] for el in dark_dict_list if
+                          abs(el['exposure'] - light_cnt_time) <= acq_time and
+                          abs(el['timestamp'] - now) <= (
+                              expire_time * 60 - acq_time)
+                          and (el['acq_time'] == acq_time)
+                          ]
```
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: single line?

File: xpdacq/xpdacq.py [hunk 13]
```diff
@@ -244,7 +247,8 @@ def open_collection(collection_name):
     # save current object
     current_name = getattr(glbl, 'collection_name', None)
     if current_name is not None:
-        with open(os.path.join(glbl.usrAnalysis_dir,current_name)+'.yaml', 'w') as f:
+        with open(os.path.join(glbl.usrAnalysis_dir, current_name) + '.yaml',
+                  'w') as f:
```
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: single line?

File: xpdacq/xpdacq.py [hunk 14]
```diff
@@ -259,15 +263,15 @@ def _insert_collection(collection_name, collection_obj, new_uid=None):
...
     glbl.collection_num = new_num
-    if new_num - ref_num >=5:
-        #print("yamlize obj, ref_num = {}, new_num = {}"
+    if new_num - ref_num >= 5:
+        # print("yamlize obj, ref_num = {}, new_num = {}"
         #      .format(ref_num, new_num))
         glbl._collection_ref_num = new_num
-        with open(os.path.join(glbl.usrAnalysis_dir,collection_name)+'.yaml', 'w') as f:
+        with open(os.path.join(glbl.usrAnalysis_dir,
+                               collection_name) + '.yaml', 'w') as f:
```
Comment (chiahaoliu) [inline xpdacq/xpdacq.py]: single line?

File: xpdacq/beamtime.py (hunk 2 above)
Comment (CJ-Wright) [inline xpdacq/beamtime.py]: It wasn't less than 72 characters when the spaces were added between the addition.

File: xpdacq/beamtime.py (hunk 3 above)
Comment (CJ-Wright) [inline xpdacq/beamtime.py]: The problem was that the liens were over the limit once the spaces were added.

File: xpdacq/calib.py (hunk 4 above)
Comment (CJ-Wright) [inline xpdacq/calib.py]: This is how PEP8 has it: http://legacy.python.org/dev/peps/pep-0008/#indentation

Comment (chiahaoliu) [inline xpdacq/calib.py]: yeah, I was citing this part

````
# Aligned with opening delimiter.
foo = long_function_name(var_one, var_two,
                         var_three, var_four)
````

Comment (CJ-Wright) [inline xpdacq/calib.py]: If that was done, then it would still be over the line.

**Comment by CJ-Wright** [comment]
@chiahaoliu Is `babble.py` still valid code? I am getting a bunch of flake8 errors from it.

File: doc/conf.py (hunk 1 above)
Comment (danielballan) [inline doc/conf.py]: +1 for not correcting autogen code

Comment (danielballan) [inline doc/conf.py]: +1 for not correcting autogen code

Comment (danielballan) [inline doc/conf.py]: +1 for not correcting autogen code

Comment (CJ-Wright) [inline doc/conf.py]: Fair enough, but why is sphinx not putting out PEP8?

File: xpdacq/utils.py (hunk 8 above)
Comment (CJ-Wright) [inline xpdacq/utils.py]: I imagine that dicts are like lists and need to be broken up.

**Comment by chiahaoliu** [comment]
@CJ-Wright  no, ``babble.py`` is not in used.

**Comment by chiahaoliu** [comment]
just test with ``unittest``, code passes with all styling change. Basically I am happy with more organized layout and would be ready to pull in.

Just out of curiosity. What is optimal balance between strict PEP 8 styling and code readability? For example, line 215 in ``beamtime.py`` doesn't seem to be very intuitive if I was completely new to this project. Any comments? @danielballan

**Comment by CJ-Wright** [comment]
The real test of its PEP8-ness is flake8 which is not passing.

**Comment by danielballan** [comment]
> What is optimal balance between strict PEP 8 styling and code readability? For example, line 215 in beamtime.py doesn't seem to be very intuitive if I was completely new to this project. 

See this introduction to PEP8 itself.

https://www.python.org/dev/peps/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds

The tl;dr is, be flexible, don't lose sight of the overall goal (readability) and absolutely never break back-compatibility over this.

When using a linter, less-than-strict PEP8 compliance is annoying, but IMO it's tolerable. An occasional  overly-long line is sometimes the more readable option. On the other hand, some projects actually enforce strict PEP* using flake8 + Travis. Matplotlib and, I believe, pandas are examples of this.

*The phrase "strict PEP8" is something of a paradox, since PEP8 itself encourages the spirit of the law over the letter of the law.

**Comment by sbillinge** [comment]
Thanks for cleaning, and @CJ-Wright thanks for the help.

If we can clean out and delete unused code, it is a Good Thing!  It will help people in the future as they try and maintain the code base.  PEP8 (spirit of, not strict) is also good in this respect.  If there is code in those unused modules, let's stash them somewhere outside the project, but let's take the time to decide if it really is useful before stashing, and delete if not.  It better be good code to save it.

**Comment by CJ-Wright** [comment]
I think this can be merged. I think there is some more cleaning left to be done (there are some unused imports and other things) but I will leave it for another day. If anyone is super interested in the issues remaining, you can look at the build [here](https://travis-ci.org/CJ-Wright/xpdAcq/jobs/160043213#L866).
Flake8 measures how well a piece of code adheres to PEP8, and although we might not be super strict that everything passes flake8, we can use it as a way to identify potential pitfalls/code to be cleaned.

File: xpdacq/beamtime.py [hunk 15]
```diff
@@ -86,43 +89,45 @@ def ct(dets, exposure, *, md=None):
...
                         'sp_uid': str(uuid.uuid4()),
                         'sp_plan_name': 'ct'})
-    plan = bp.count([glbl.area_det], md = _md)
+    plan = bp.count([glbl.area_det], md=_md)
     plan = bp.subs_wrapper(plan, LiveTable([glbl.area_det]))
     yield from plan
 
 
-def Tramp(dets, exposure, Tstart, Tstop, Tstep, *, md=None):
+def Tramp(dets, exposure, tstart, tstop, tstep, *, md=None):
```
Comment (chiahaoliu) [inline xpdacq/beamtime.py]: Just found this.

As styling, I understand capitalized variable name is not ideal but ``T`` here implies temperature, which is usually capitalized in science field. Can we keep it?

Comment (danielballan) [inline xpdacq/beamtime.py]: +1 for leaving it as is. See "good reasons to ignore a particular guideline" section of PEP8.

**Comment by sbillinge** [comment]
yup, we want that capitalized

File: xpdacq/beamtime.py (hunk 15 above)
Comment (CJ-Wright) [inline xpdacq/beamtime.py]: Reverted

**Pull request merged**